client-app/src/pages/account/views/AccountInfo.vue

```diff
@@ -36,21 +37,35 @@
 <script lang="ts">
 import Vue from "vue";
 import Component from "vue-class-component";
-import { FETCH_PROFILE, UPDATE_USER } from "@account/store/definitions";
+import { FETCH_PROFILE, UPDATE_USER } from "@account/store/modules/profile/definitions";
+import { State, Getter, Action, Mutation, namespace } from 'vuex-class';
+import { User, UserUpdateInfo } from "@common/api/api-clients";
+
+
+const profileModule = namespace('profileModule');
 
 @Component
 export default class AccountInfo extends Vue {
-  get profile() {
-    return this.$store.getters.profile;
-  }
-  get isLoading() {
-    return this.$store.getters.isLoading;
-  }
+  @profileModule.Getter('profile')
+  profile!: User;
+  @profileModule.Getter('isLoading')
+  isLoading!: boolean;
+
+  @profileModule.Action(FETCH_PROFILE)
+  fetchProfile!: () => void;
+  @profileModule.Action(UPDATE_USER)
+  updateProfile!: ( payload: { userUpdateInfo: UserUpdateInfo }) => void;
+
   mounted() {
-    this.$store.dispatch(FETCH_PROFILE);
+    this.fetchProfile();
   }
   update() {
-    this.$store.dispatch(UPDATE_USER, this.profile);
+    const userUpdateInfo =  new UserUpdateInfo();
```

Comment:
why do we need this?
if it really necessary why w couldn't fo it with using spread operator?

Reply:
Because property roles have a different type.